src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2459,6 +2462,15 @@ def _handle_addons_args(cmd, addons_str, subscription_id, resource_group_name, a
     if 'azure-policy' in addons:
         addon_profiles['azurepolicy'] = ManagedClusterAddonProfile(enabled=True)
         addons.remove('azure-policy')
+    if 'virtual-node' in addons:
+        if not aci_subnet_name or not vnet_subnet_id:
+            raise CLIError('"--enable-addons virtual-node" requires "--aci-subnet-name" and "--vnet-subnet-id".')
+        # TODO: how about aciConnectorwindows, what is its addon name?
+        addon_profiles['aciConnectorLinux'] = ManagedClusterAddonProfile(
```

Comment:
I think it's the 'aciConnector' addon profile, not 'aciConnectorLinux' we need to use. See: https://github.com/Azure/azure-cli/blob/0c88076fafb279665529bfa8178e76eab7c2bbfe/src/azure-cli/azure/cli/command_modules/acs/custom.py#L1999 and the implementation in the extension CLI: https://github.com/Azure/azure-cli-extensions/blob/3c7edde5e5475a9c1f2a8e9b602923da432d6f93/src/aks-preview/azext_aks_preview/custom.py#L2513

Reply:
Thanks @MikkelHegn, it is the profile should be `aciConnectorLinux` for Linux, see `addon += os_type` in https://github.com/Azure/azure-cli/blob/0c88076fafb279665529bfa8178e76eab7c2bbfe/src/azure-cli/azure/cli/command_modules/acs/custom.py#L2369-L2371, which is currently defined in aks-rp. 
Currently It is name is only `aciConnector`, which would be used for both Linux and Windows, and that is why I am asking what is its name for Windows.

Reply:
Windows is not supported yet, but I think `addon += os_type` should also work when we add Windows support in the future.

Reply:
yes, `aciConnector`  equals to `aciConnectorLinux`  today, and we are enabling it with param `--enable-addon virtual-node`.
For further view, what if we enable it for Windows in the future? should it be `--enable-addon virtual-node-windows` ?